gensim/models/keyedvectors.py

```diff
@@ -1695,6 +1697,79 @@ def intersect_word2vec_format(self, fname, lockf=0.0, binary=False, encoding='ut
             msg=f"merged {overlap_count} vectors into {self.vectors.shape} matrix from {fname}",
         )
 
+    def vectors_for_all(self, keys: Union[Iterable, Dictionary],
+                        allow_inference: bool = True,
+                        copy_vecattrs: bool = False) -> 'KeyedVectors':
+        """Produce vectors for all given keys as a new :class:`KeyedVectors` object.
+
+        Notes
+        -----
+        The keys will always be deduplicated. For optimal performance, you should not pass entire
+        corpora to the method. Instead, you should construct a
+        :class:`~gensim.corpora.dictionary.Dictionary` object out of your corpus first:
+
+        >>> import gensim.downloader as api
```

Comment:
Because I consider the post-installation loading & execution of arbitrary new, unversioned source code to be a dangerously reckless process we should eliminate, for the reasons in #2283, I never like to see the use of `gensim.downloader` encouraged.

Reply:
The use of `gensim.downloader` is pervasive throughout auto-examples and docstrings. I would suggest making a batch change to all of these when there is an agreement on how to resolve #2283. What would you suggest?

Reply:
For the sake of simplicity, if it's possible to avoid using the downloader by using an inline corpus, or something readily available, then the documentation would be better without the downloader.

On top of the problems that @gojomo mentioned, using the downloader instantly makes the docstring non-doctest-able, because downloading those datasets can take a long time.

Reply:
I removed the Gensim downloader from the examples in 32c14c5d7ecc0577bfc59e1c3cf9e96bbb3b9bc9 and 712ee6164e18215092859d5e4611d8ccb06578a9.